nncf/tensorflow/accuracy_aware_training/runner.py

```diff
@@ -13,132 +13,104 @@
+    def __init__(self, accuracy_aware_training_params, verbose=True, dump_checkpoints=True, lr_updates_needed=True,
+                 minimal_compression_rate=0.05, maximal_compression_rate=0.95):
+        super().__init__(accuracy_aware_training_params, verbose, dump_checkpoints, lr_updates_needed,
+                         minimal_compression_rate=minimal_compression_rate,
+                         maximal_compression_rate=maximal_compression_rate)
 
     def dump_checkpoint(self, model, compression_controller):
-        checkpoint_path = osp.join(self._checkpoint_save_dir, 'acc_aware_checkpoint_last.pb')
-        model.save_weights(checkpoint_path)
-
-        if self.best_val_metric_value == self.current_val_metric_value:
-            best_path = osp.join(self._checkpoint_save_dir,
-                                 'acc_aware_checkpoint_best_compression_rate_'
-                                 '{comp_rate:.3f}.ckpt'.format(comp_rate=self.compression_rate_target))
-            self._best_checkpoints[self.compression_rate_target] = best_path
-            model.save_weights(best_path)
-
-    def load_best_checkpoint(self, model):
-        # load checkpoint with highest compression rate and positive acc budget
-        possible_checkpoint_rates = [comp_rate for (comp_rate, acc_budget) in self._compressed_training_history
-                                     if acc_budget >= 0]
-        if not possible_checkpoint_rates:
-            nncf_logger.warning('Could not produce a compressed model satisfying the set accuracy '
-                                'degradation criterion during training. Increasing the number of training '
-                                'epochs')
-        best_checkpoint_compression_rate = sorted(possible_checkpoint_rates)[-1]
-        resuming_checkpoint_path = self._best_checkpoints[best_checkpoint_compression_rate]
-        nncf_logger.info('Loading the best checkpoint found during training '
-                         '{}...'.format(resuming_checkpoint_path))
-        model.load_weights(resuming_checkpoint_path)
-
-    @property
-    def compressed_training_history(self):
-        return self._compressed_training_history
+        # a workaround because for tensorflow backend disabling compression scheduler does not work properly
+        is_best_checkpoint = (self.best_val_metric_value == self.current_val_metric_value and
+                              isinstance(compression_controller.scheduler, StubCompressionScheduler))
+        if not self.dump_checkpoints and not is_best_checkpoint:
+            return
+
+        if self._dump_checkpoint_fn is not None:
+            checkpoint_path = self._dump_checkpoint_fn(model, compression_controller, self, self._checkpoint_save_dir)
+        else:
+            checkpoint_path = osp.join(self._checkpoint_save_dir,
+                                       f'acc_aware_checkpoint_last{self.checkpoint_path_extension}')
```

Comment:
Refactor to use `pathlib`

Reply:
It was decided to improve code quality within a separate task

Reply:
It was decided to improve code quality within a separate task